include/z64camera.h

```diff
@@ -0,0 +1,403 @@
+    /* 0x1D */ CAM_FUNC_BATT3,
+    /* 0x1E */ CAM_FUNC_BATT4,
+    /* 0x1F */ CAM_FUNC_FIXD0,
+    /* 0x20 */ CAM_FUNC_FIXD1,
+    /* 0x21 */ CAM_FUNC_FIXD2,
+    /* 0x22 */ CAM_FUNC_FIXD3,
+    /* 0x23 */ CAM_FUNC_FIXD4,
+    /* 0x24 */ CAM_FUNC_DATA0,
+    /* 0x25 */ CAM_FUNC_DATA1,
+    /* 0x26 */ CAM_FUNC_DATA2,
+    /* 0x27 */ CAM_FUNC_DATA3,
+    /* 0x28 */ CAM_FUNC_DATA4,
+    /* 0x29 */ CAM_FUNC_UNIQ0,
+    /* 0x2A */ CAM_FUNC_UNIQ1,
+    /* 0x2B */ CAM_FUNC_UNIQ2,
+    /* 0x2C */ CAM_FUNC_UNIQ3,
+    /* 0x2D */ CAM_FUNC_UNIQ4,
+    /* 0x2E */ CAM_FUNC_UNIQ5,
+    /* 0x2F */ CAM_FUNC_UNIQ6,
+    /* 0x30 */ CAM_FUNC_UNIQ7,
+    /* 0x31 */ CAM_FUNC_UNIQ8,
+    /* 0x32 */ CAM_FUNC_UNIQ9,
+    /* 0x33 */ CAM_FUNC_DEMO0,
+    /* 0x34 */ CAM_FUNC_DEMO1,
+    /* 0x35 */ CAM_FUNC_DEMO2,
+    /* 0x36 */ CAM_FUNC_DEMO3,
+    /* 0x37 */ CAM_FUNC_DEMO4,
+    /* 0x38 */ CAM_FUNC_DEMO5,
+    /* 0x39 */ CAM_FUNC_DEMO6,
+    /* 0x3A */ CAM_FUNC_DEMO7,
+    /* 0x3B */ CAM_FUNC_DEMO8,
+    /* 0x3C */ CAM_FUNC_DEMO9,
+    /* 0x3D */ CAM_FUNC_SPEC0,
+    /* 0x3E */ CAM_FUNC_SPEC1,
+    /* 0x3F */ CAM_FUNC_SPEC2,
+    /* 0x40 */ CAM_FUNC_SPEC3,
+    /* 0x41 */ CAM_FUNC_SPEC4,
+    /* 0x42 */ CAM_FUNC_SPEC5,
+    /* 0x43 */ CAM_FUNC_SPEC6,
+    /* 0x44 */ CAM_FUNC_SPEC7,
+    /* 0x45 */ CAM_FUNC_SPEC8,
+    /* 0x46 */ CAM_FUNC_SPEC9,
+    /* 0x47 */ CAM_FUNC_MAX
+} CameraFuncType;
+
+typedef enum {
+    /* 0x00 */ CAM_RO_DATA_00,
+    /* 0x01 */ CAM_RO_DATA_01,
+    /* 0x02 */ CAM_RO_DATA_02,
+    /* 0x03 */ CAM_RO_DATA_03,
+    /* 0x04 */ CAM_RO_DATA_04,
+    /* 0x05 */ CAM_RO_DATA_05,
+    /* 0x06 */ CAM_RO_DATA_06,
+    /* 0x07 */ CAM_RO_DATA_07,
+    /* 0x08 */ CAM_RO_DATA_08,
+    /* 0x09 */ CAM_RO_DATA_09,
+    /* 0x0A */ CAM_RO_DATA_10,
+    /* 0x0B */ CAM_RO_DATA_11,
+    /* 0x0C */ CAM_RO_DATA_12,
+    /* 0x0D */ CAM_RO_DATA_13,
+    /* 0x0E */ CAM_RO_DATA_14,
+    /* 0x0F */ CAM_RO_DATA_15,
+    /* 0x10 */ CAM_RO_DATA_16,
+    /* 0x11 */ CAM_RO_DATA_17,
+    /* 0x12 */ CAM_RO_DATA_18,
+    /* 0x13 */ CAM_RO_DATA_19,
+    /* 0x14 */ CAM_RO_DATA_20,
+    /* 0x15 */ CAM_RO_DATA_21,
+} CameraReadOnlyDataType;
+
+typedef struct {
+    /* 0x0 */ s16 val;
+    /* 0x2 */ s16 param;
+} CameraModeValue; // size = 0x4
+
+typedef struct {
+    /* 0x0 */ s16 funcId;
+    /* 0x2 */ s16 numValues;
+    /* 0x4 */ CameraModeValue* values;
+} CameraMode; // size = 0x8
+
+/**
+ * Flags:
+ * (flags & 0xF): Priority (lower value has higher priority)
+ * (flags & 0x40000000): Store previous setting and bgCamData, also ignores water checks
+ * (flags & 0x80000000): Set camera setting based on bg/scene data and reset action function state
+ */
+typedef struct {
+    /* 0x0 */ u32 validModes;
+    /* 0x4 */ u32 flags;
+    /* 0x8 */ CameraMode* cameraModes;
+} CameraSetting; // size = 0xC
+
+typedef struct Camera {
+    /* 0x000 */ char paramData[0x50]; // function Data, acts like a heap that's reset every time a new action function is switched to
```

Comment:
Is this actual `char` or should it be `u8`?

Reply:
There's a lot going on with the first 80 bytes, paramData, see this OoT PR for just a taste:
https://github.com/zeldaret/oot/pull/1108

But yes, I suppose it should just be u8